utils/difference.js

```diff
@@ -0,0 +1,61 @@
+const { executeWithValues: executeFormula } = require('./formula');
+
+/*
+ * Do calculations for all difference_* values for the given row
+ * @param{Object} row - The row to do calculations for
+ */
+function doDifferenceCalculations(row) {
+  calculateDifferences(row);
+  calculateDifferencePercents(row);
+}
+
+/*
+ * Calculate difference_sum, difference_m, and significant
+ * @param{row} - The row to do calculations for
+ */
+function calculateDifferences(row) {
+  const updatedRow = row;
+  if (exists(updatedRow.sum) && exists(updatedRow.comparison_sum)) {
+    updatedRow.difference_sum = executeFormula('delta', [updatedRow.sum, updatedRow.comparison_sum]);
+
+    if (exists(updatedRow.m) && exists(updatedRow.comparison_m)) {
+      updatedRow.difference_m = executeFormula('delta_m', [updatedRow.m, updatedRow.comparison_m]);
+
+      // TODO rename difference_significant
+      if (updatedRow.difference_sum !== 0) updatedRow.significant = executeFormula('significant', [updatedRow.difference_sum, updatedRow.difference_m]);
+    }
+  }
+
+  // special handling for 'decennial' rows, which do not have MOE and are all considered 'significant'
+  if (row.profile === 'decennial') row.significant = true;
+}
+
+/*
+ * Calculate difference_percent, difference_percent_m, and percent_significant
+ * @param{row} - The row to do calculations for
+ */
+function calculateDifferencePercents(row) {
+  const updatedRow = row;
+  if (exists(updatedRow.percent) && exists(updatedRow.comparison_percent)) {
+    const differencePercent = executeFormula('delta', [updatedRow.percent * 100, updatedRow.comparison_percent * 100]);
```

Comment:
These parametric formula templates are awesome - it feels like this particular arguments should be handled in the formula itself no?

In other words, _what should be handled in JavaScript_ and _what should be handled in the Excel formula parser_? How to make that distinction?

Reply:
I did this check outside of the formula in order to be able to reuse the same 'delta' formula that I use in all the other calculations; again if we think it's clearer to do it in a formula it would be very easy to add another formula to this list and use that here !